indolaw-parser/parser_utils.py

```diff
@@ -552,8 +552,35 @@ def print_around(law: List[str], i: int) -> None:
     ''')
 
 
-def convert_tree_to_json(node: Union[ComplexNode, PrimitiveNode]) -> Dict[str, Any]:
+def convert_tree_to_json(node: Union[ComplexNode, PrimitiveNode], ketentuan_umum_list: List[str]) -> Dict[str, Any]:
     if isinstance(node, PrimitiveNode):
+        if get_parent_node(node, Structure.BAB) is not None and node.type == Structure.PLAINTEXT:
+            for title in ketentuan_umum_list:
+                index = node.text.upper().find(title.upper())
+
+                pasal_node = get_parent_node(node, Structure.PASAL)
+                is_definition = get_id(pasal_node) == 'pasal-1'
+
+                if is_word_part_of_text(node.text, title, index) and not is_definition:
+                    text = node.text[index:index+len(title)]
+                    '''
+                    Determine if the word is actually a part of a bigger definition, 
+                        and if such case happens, opt for the longer word
+
+                    Ex: 
+                    assume "Lorem ipsum" and "ipsum" exist in the dictionary
+
+                    "Lorem ipsum dolor sit amet" -> "${Lorem ipsum} dolor sit amet"
+                    "Ipsum dolor sit amet" -> "${Ipsum} dolor sit amet"
+                    "Lorem ipsum ipsum lorem" -> "${Lorem ipsum} ${ipsum} lorem"
+                    '''
+
+                    match = re.findall(f'.*\${{.*{text}.*}}.*', node.text)
+                    is_part_of_other_definition = len(match) > 0
+
+                    if not is_part_of_other_definition:
+                        node.text = node.text.replace(text, f'${{{text}}}')
```

Comment:
this looks good to me, but what's the pro/cons of this approach vs. having a separate dictionary that maybe stores the start/end indexes instead of modifying the text itself? just curious if you've thought about it

Reply:
Tbh, I haven't thought about that, but one minor con that I have in mind is that the checking would have to include a scan of all the values in the dictionary (to ensure that it's not a part of an existing definition). But, will try to consider this implementation as well